include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -82,6 +82,170 @@ namespace internal
     {}
 
 
+
+    namespace
+    {
+      template <int dim, int spacedim>
+      unsigned int
+      setup_parallel_data_base_internal(
+        const unsigned int                                  n_subdivisions,
+        const dealii::hp::MappingCollection<dim, spacedim> &mapping_collection,
+        const std::vector<
+          std::shared_ptr<dealii::hp::FECollection<dim, spacedim>>>
+          &               finite_elements,
+        const UpdateFlags update_flags,
+        const bool        use_face_values,
+        std::vector<std::shared_ptr<dealii::hp::FEValues<dim, spacedim>>>
+          &x_fe_values,
+        std::vector<std::shared_ptr<dealii::hp::FEFaceValues<dim, spacedim>>>
+          &x_fe_face_values)
+      {
+        unsigned int n_q_points = 0;
+        if (use_face_values == false)
+          {
+            // determine if specific quadrature rules need to set up
+            bool needs_hypercube_setup = false;
+            bool needs_simplex_setup   = false;
+            bool needs_wedge_setup     = false;
+
+            for (const auto &fe : finite_elements)
+              for (unsigned int i = 0; i < fe->size(); ++i)
+                switch ((*fe)[i].reference_cell_type())
+                  {
+                    case ReferenceCell::Type::Vertex:
+                    case ReferenceCell::Type::Line:
+                    case ReferenceCell::Type::Quad:
+                    case ReferenceCell::Type::Hex:
+                      needs_hypercube_setup |= true;
+                      break;
```

Comment:
I feel like this all could have been done more easily using a `std::any` call with a nice little lambda function. That could also have been done in the rhs of the initialization of these variables above, and then you can make them `const`